pulsar-broker/src/test/java/com/yahoo/pulsar/client/api/SimpleProducerConsumerStatTest.java

```diff
@@ -323,6 +327,40 @@ public void testSendTimeout(int batchMessageDelayMs) throws Exception {
         assertEquals(cStat.getTotalMsgsReceived(), cStat.getTotalAcksSent());
         log.info("-- Exiting {} test --", methodName);
     }
+    
+    @Test
+    public void testBatchMessagesRateOut() throws PulsarClientException, InterruptedException, PulsarAdminException {
+        log.info("-- Starting {} test --", methodName);
+        String topicName = "persistent://my-property/cluster/my-ns/testBatchMessagesRateOut";
+        double produceRate = 17;
+        ConsumerConfiguration consumerConf = new ConsumerConfiguration();
+        consumerConf.setSubscriptionType(SubscriptionType.Exclusive);
+        Consumer consumer = pulsarClient.subscribe(topicName, "my-subscriber-name", consumerConf);
+        ProducerConfiguration producerConf = new ProducerConfiguration();
+        producerConf.setBatchingMaxMessages(5);
+        producerConf.setBatchingEnabled(true);
+        producerConf.setBatchingMaxPublishDelay(2, TimeUnit.SECONDS);
+
+        Producer producer = pulsarClient.createProducer(topicName, producerConf);
+        AtomicBoolean runTest = new AtomicBoolean(true);
+        Thread t1 = new Thread(() -> {
+            RateLimiter r = RateLimiter.create(produceRate);
+            while (runTest.get()) {
+                r.acquire();
+                producer.sendAsync("Hello World".getBytes());
+                consumer.receiveAsync().thenAccept(message -> consumer.acknowledgeAsync(message));
+            }
+        });
+        t1.start();
+        Thread.sleep(2000); // Two seconds sleep
+        pulsar.getBrokerService().updateRates();
+        double actualRate = admin.persistentTopics().getStats(topicName).msgRateOut;
+        log.info("actualRate : {}, produceRate : {}", actualRate, produceRate);
+        assertTrue((produceRate - 1) <= actualRate && actualRate <= (produceRate + 1));
```

Comment:
@jai1  Can you run this test multiple time to ensure is not intermittently failing?